packages/autorest.python/autorest/codegen/templates/serialization.py.jinja2

```diff
@@ -752,9 +752,10 @@ class Serializer(object):
             if data_type.startswith("["):
                 internal_data_type = data_type[1:-1]
                 data = [self.serialize_data(d, internal_data_type, **kwargs) if d is not None else "" for d in data]
+                data = ["" if s is None else str(s) for s in data]
                 if not kwargs.get("skip_quote", False):
                     data = [quote(str(d), safe="") for d in data]
-                return str(self.serialize_iter(data, internal_data_type, **kwargs))
+                return kwargs["div"].join(data)
```

Comment:
This change aims to prevent duplicated serialization for list elements.

Reply:
@iscai-msft @lmazuel @msyyc please check this change since it will result in generated code change in all kinds of generated SDKs.

Reply:
The change may be a little dangerous since it changes both mgmt and DPG. Do we have specific test about the change?

Reply:
I've created another issue and will solve this serialization problem in another PR to unblock this PR's merge.

Reply:
@msyyc the actual change and test could be found here: https://github.com/Azure/msrest-for-python/pull/259

Reply:
Tests has already been migrated to this repo.